crates/ruff_python_formatter/src/lib.rs

```diff
@@ -167,6 +167,21 @@ pub fn format_node<'a>(
     Ok(formatted)
 }
 
+/// Public function for generating a printable string of the debug comments.
+pub fn pretty_comments(formatted: &Formatted<PyFormatContext>, source: &str) -> String {
+    let comments = formatted
+        .context()
+        .comments()
+        .debug(SourceCode::new(source));
+
+    // When comments are empty we'd display an empty map '{}'
+    if comments.empty() {
+        String::new()
+    } else {
+        std::format!("{comments:#?}")
+    }
```

Comment:
![image](https://github.com/astral-sh/ruff/assets/14341145/1ebf9505-d38f-44ce-9e04-4798fdc2c338)

Reply:
That'd be the expected output as-is (with no condition on empty comments). It was a nit of mine so I decided to print nothing if comments were 'empty'.